schoe14/english-mastery-app
# Add OAuth2 and social account signup for users

Summary
Adds OAuth2 authentication to improve the security of the application. Eliminates the need for password-based logins.
Adds Google Sign Up.

Background
The current authentication system relies on username and password login, which is less secure and provides a poor user experience.

Implementation Details
Integrated the Django OAuth2 library
Added new OAuth2 settings in the settings.py
Created new authentication views

How to Test
Navigate to the login page
Click on "Sign in with Google"
Complete the OAuth2 flow
You should be redirected back to the homepage, now logged in

Caveats
The current implementation only supports Google OAuth2. Future iterations could include more providers.
